pkg/descheduler/strategies/duplicates.go

```diff
@@ -166,3 +145,20 @@ func hasExcludedOwnerRefKind(ownerRefs []metav1.OwnerReference, strategy api.Des
 	}
 	return false
 }
+
+func getUniqueOwnerKeyOfPod(pod *v1.Pod) string {
+	ownerRefList := podutil.OwnerRef(pod)
+	podContainerKeys := make([]string, 0, len(ownerRefList)*len(pod.Spec.Containers))
+	for _, ownerRef := range ownerRefList {
+		for _, container := range pod.Spec.Containers {
+			// Namespace/Kind/Name should be unique for the cluster.
+			// We also consider the image, as 2 pods could have the same owner but serve different purposes
+			// So any non-unique Namespace/Kind/Name/Image pattern is a duplicate pod.
+			s := strings.Join([]string{pod.ObjectMeta.Namespace, ownerRef.Kind, ownerRef.Name, container.Image}, "/")
+			podContainerKeys = append(podContainerKeys, s)
+		}
+	}
+	sort.Strings(podContainerKeys)
+	// Each pod has a list of owners and a list of containers, and each container has 1 image spec.
+	return podContainerKeys[0]
```

Comment:
Counter-example:
- pod P1 has two containers C1 and C2 (each with a different image)
- pod P2 has two containers C3 and C4 (each with a different image).
- containers C2 and C4 have the same image
- images of containers C1, C2 and C3 are unique
- sorting containers key gives the following order: C2>C1, C4>C3.
Then:
- getUniqueOwnerKeyOfPod(P1) = C2
- getUniqueOwnerKeyOfPod(P2) = C4

Both pods P1 and P2 are not duplicates (since C1 != C3). However, `getUniqueOwnerKeyOfPod` marks both pods as duplicates.

Reply:
@ingvagabund could you please elaborate on how you determined `getUniqueOwnerKeyOfPod` marks P1 & P2 as duplicates? 


> getUniqueOwnerKeyOfPod(P1) = C2
> getUniqueOwnerKeyOfPod(P2) = C4
> 

how does this assertion imply P1 and P2 are duplicates?

Reply:
> how does this assertion imply P1 and P2 are duplicates? wouldn't they both return the same value if it were?

`getUniqueOwnerKeyOfPod` takes the first item in sorted `podContainerKeys` list. Which means the pod container keys that make both pods `P1` and `P2` distinct may get ignored based on how `podContainerKeys` list is sorted.

Reply:
Assuming both pods `P1` and `P2` are in the same namespace `NS`, owned by the same owner of kind `K` and name `N`. 

For `P1`: `podContainerKeys` list has `[NS/K/N/C1image, NS/K/N/C2image]`.
For `P2`: `podContainerKeys` list has `[NS/K/N/C3image, NS/K/N/C4image]`.

> sorting containers key gives the following order: C2>C1, C4>C3.

After sorting, you will get:

For `P1`: `podContainerKeys` list has `[NS/K/N/C2image, NS/K/N/C1image]`.
For `P2`: `podContainerKeys` list has `[NS/K/N/C4image, NS/K/N/C3image]`.

Given `getUniqueOwnerKeyOfPod` picks the first item in the list, then:

`getUniqueOwnerKeyOfPod(P1)` = `NS/K/N/C2image`
`getUniqueOwnerKeyOfPod(P2)` = `NS/K/N/C4image`

Given both `C2image` and `C4image` images are identical, both pods `P1` and `P2` get the same unique key.

Reply:
Thanks that makes sense now

Reply:
@ingvagabund apologies for not getting back to you sooner. I have to admit i didn't understand it fully, but i finally found the real question i needed to ask.

If `C2` and `C4` have the same image, why do they have different image names?

```
getUniqueOwnerKeyOfPod(P1) = NS/K/N/C2image
getUniqueOwnerKeyOfPod(P2) = NS/K/N/C4image
```

From your example, two images can have different image names, but at the same time are identical. So what makes an image unique?

Reply:
> containers C2 and C4 have the same image

From that C2image = C4image. `C2image`, resp. `C4image` is a variable name. Though, their value is the same.

Reply:
Ah ok, thanks. let me test that.